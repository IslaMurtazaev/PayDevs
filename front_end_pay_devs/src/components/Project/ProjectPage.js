import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { projectActions } from "../../actions/project";

class ProjectPage extends Component {
  componentWillMount() {
    if (!this.props.projects.length) this.props.getAllProjects();
  }

  render() {
    const { user } = this.props.user; // why we are wrapping user inside a user?
    const { projects } = this.props;
    return (
      <div>
        <div className="user">
          {user && <div className="username">{user.username}</div>}
          {user && <div className="email">{user.email}</div>}
        </div>
        <Link className="newProjectLink" to="/project/create">
          New Project
        </Link>
        <div className="projectList">
          <ul>
            {projects.map(project => (
              <li key={project.id}>
                <div className="projectInfo">
                  <Link className="projectLink" to={`/project/${project.id}`}>
                    {project.title}
                  </Link>
                  <p>{new Date(project.start_date).toDateString()}</p>
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

export default connect(
  state => ({
    user: state.user,
    projects: state.projects
  }),
  dispatch => ({
    getAllProjects: () => {
      dispatch(projectActions.getAll());
    }
  })
)(ProjectPage);
