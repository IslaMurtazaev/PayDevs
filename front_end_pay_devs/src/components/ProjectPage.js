import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { projectActions } from "../actions/project";

class ProjectPage extends Component {
  onClick() {
    this.props.onGetAllProjects();
  }

  componentDidMount() {
    this.props.onGetAllProjects();
  }

  render() {
    const { user } = this.props.user;
    const { projects } = this.props;

    return (
      <div>
        {user && <div> Username: {user.username}</div>}
        {user && <div> Email: {user.email}</div>}
        {user && <Link to="/login">Logout</Link>}
        <ul>
          {projects.map(project => (
            <li key={project.id}>
              <Link to={`/project/${project.id}`}>{project.title}</Link>
            </li>
          ))}
        </ul>
        <Link to={"/project/create"}>New project</Link>
      </div>
    );
  }
}

export default connect(
  state => ({
    user: state.user,
    projects: state.projects
  }),
  dispatch => ({
    onGetAllProjects: () => {
      dispatch(projectActions.getAll());
    },
    onClearProjects: () => {
      dispatch(projectActions.clearAll());
    }
  })
)(ProjectPage);
