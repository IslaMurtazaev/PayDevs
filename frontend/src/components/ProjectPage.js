import React, { Component } from "react";
import { Link } from "react-router-dom";

class ProjectPage extends Component {
  componentDidMount() {
    this.props.getAllProjects();
  }

  render() {
    const { user } = this.props.user;
    const { projects } = this.props;

    return (
      <div>
        <div className="user">
          {user && <div className="username">{user.username}</div>}
          {user && <div className="email">{user.email}</div>}
        </div>
        <Link className="newProjectLink" to="/project/create">
          New Project
        </Link>
        <div className="projectList">
          <ul>
            {projects.map(project => (
              <li key={project.id}>
                <div className="projectInfo">
                  <Link className="projectLink" to={`/project/${project.id}`}>
                    {project.title}
                  </Link>
                  <p>{new Date(project.start_date).toDateString()}</p>
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

export default ProjectPage;
