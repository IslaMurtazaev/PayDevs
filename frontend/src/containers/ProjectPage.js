import { connect } from "react-redux";

import { projectActions } from "../actions/project";
import ProjectPage from "../components/ProjectPage";

export default connect(
  state => ({
    user: state.user,
    projects: state.projects
  }),
  dispatch => ({
    getAllProjects: () => {
      dispatch(projectActions.getAll());
    }
  })
)(ProjectPage);
