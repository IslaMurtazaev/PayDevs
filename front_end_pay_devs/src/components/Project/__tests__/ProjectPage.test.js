import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json"
jest.mock("../../../index.js", () => require("history"));

import ProjectPage from "../ProjectPage";

describe("<ProjectItem />", () => {
  const initialState = {
    user: {
      user: {
        username: "IslaMurtazaev",
        email: "islam.muratazaev@gmail.com"
      }
    },
    projects: [
      {
        id: 1,
        title: "PayDevs",
        description: "blablabla",
        start_date: "2018-08-24T00:00:00.000Z",
        end_date: "2018-09-30T10:00:00.000Z",
        type_of_payment: "M_P",
        status: true
      },
      {
        id: 2,
        title: "PayDevs2",
        description: "blablabla2",
        start_date: "2018-08-24T00:00:00.000Z",
        end_date: "2018-09-30T10:00:00.000Z",
        type_of_payment: "H_P",
        status: false
      }
    ]
  };
  const mockStore = configureStore();

  let store = mockStore(initialState);
  let container = shallow(<ProjectPage store={store} />).dive();

  it("renders username based on props", () => {
    expect(container.find(".username").text()).toBe("IslaMurtazaev");
  });

  it("renders email based on props", () => {
    expect(container.find(".email").text()).toBe("islam.muratazaev@gmail.com");
  });

  it("matches previous snap", () => {
    expect(toJson(container)).toMatchSnapshot();
  })
});
