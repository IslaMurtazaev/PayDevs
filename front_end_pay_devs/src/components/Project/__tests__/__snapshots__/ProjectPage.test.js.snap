// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectPage /> matches previous snap 1`] = `
<div>
  <div
    className="user"
  >
    <div
      className="username"
    >
      IslaMurtazaev
    </div>
    <div
      className="email"
    >
      islam.muratazaev@gmail.com
    </div>
  </div>
  <Link
    className="newProjectLink"
    replace={false}
    to="/project/create"
  >
    New Project
  </Link>
  <div
    className="projectList"
  >
    <ul>
      <li
        key="1"
      >
        <div
          className="projectInfo"
        >
          <Link
            className="projectLink"
            replace={false}
            to="/project/1"
          >
            PayDevs
          </Link>
          <p>
            Fri Aug 24 2018
          </p>
        </div>
      </li>
      <li
        key="2"
      >
        <div
          className="projectInfo"
        >
          <Link
            className="projectLink"
            replace={false}
            to="/project/2"
          >
            PayDevs2
          </Link>
          <p>
            Fri Aug 24 2018
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
